src/wasm/wasm-validator.cpp

```diff
@@ -237,14 +237,17 @@ void WasmValidator::visitAtomicRMW(AtomicRMW* curr) {
 void WasmValidator::visitAtomicCmpxchg(AtomicCmpxchg* curr) {
   validateMemBytes(curr->bytes, curr->type, curr);
 }
-void WasmValidator::validateMemBytes(uint8_t bytes, WasmType ty, Expression* curr) {
+void WasmValidator::validateMemBytes(uint8_t bytes, WasmType type, Expression* curr) {
+  if (type == unreachable) {
```

Comment:
Right, that's why I suggested `shouldBeEqualOrFirstIsUnreachable` which is what we want for that. We can still compare bytes to 1,2, and 4. My point is that a 5-byte load is always invalid.